pkg/controllers/readiness/wellknown_ready_controller.go

```diff
@@ -0,0 +1,351 @@
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	"github.com/openshift/cluster-authentication-operator/pkg/transport"
+)
+
+var kasServicePort int
+
+func init() {
+	(&sync.Once{}).Do(func() {
+		var err error
+		kasServicePort, err = strconv.Atoi(os.Getenv("KUBERNETES_SERVICE_PORT_HTTPS"))
+		if err != nil {
+			klog.Warningf("Defaulting KUBERNETES_SERVICE_PORT_HTTPS to 443 due to parsing error: %v", err)
+			kasServicePort = 443
+		}
+	})
+}
+
+type wellKnownReadyController struct {
+	serviceLister  corev1lister.ServiceLister
+	endpointLister corev1lister.EndpointsLister
+	operatorClient v1helpers.OperatorClient
+	authLister     configv1lister.AuthenticationLister
+	routeLister    routev1lister.RouteLister
+}
+
+// knownConditionNames lists all condition types used by this controller.
+// These conditions are operated and defaulted by this controller.
+// Any new condition used by this controller sync() loop should be listed here.
+var knownConditionNames = sets.NewString(
+	"WellKnownEndpointDegraded",
+	"WellKnownRouteDegraded",
+	"WellKnownNotReadyProgressing",
+	"WellKnownNotReadyAvailable",
+)
+
+func NewWellKnownReadyController(kubeInformersNamespaced informers.SharedInformerFactory, configInformers configinformer.SharedInformerFactory, routeInformer routeinformer.RouteInformer,
+	operatorClient v1helpers.OperatorClient, recorder events.Recorder) factory.Controller {
+	c := &wellKnownReadyController{
+		serviceLister:  kubeInformersNamespaced.Core().V1().Services().Lister(),
+		endpointLister: kubeInformersNamespaced.Core().V1().Endpoints().Lister(),
+		authLister:     configInformers.Config().V1().Authentications().Lister(),
+		routeLister:    routeInformer.Lister(),
+		operatorClient: operatorClient,
+	}
+
+	return factory.New().ResyncEvery(30*time.Second).WithInformers(
+		kubeInformersNamespaced.Core().V1().Services().Informer(),
+		kubeInformersNamespaced.Core().V1().Endpoints().Informer(),
+		configInformers.Config().V1().Authentications().Informer(),
+		routeInformer.Informer(),
+	).WithSync(c.sync).ToController("WellKnownReadyController", recorder.WithComponentSuffix("wellknown-ready-controller"))
+}
+
+func (c *wellKnownReadyController) sync(ctx context.Context, controllerContext factory.SyncContext) error {
+	foundConditions := []operatorv1.OperatorCondition{}
+
+	authConfig, configConditions := c.getAuthConfig()
+	foundConditions = append(foundConditions, configConditions...)
+
+	route, routeConditions := c.getRoute()
+	foundConditions = append(foundConditions, routeConditions...)
+
+	if authConfig != nil && route != nil {
+		// TODO: refactor this to return conditions
+		ready, endpointsMessage, err := c.isWellknownEndpointsReady(authConfig, route)
+		if !ready {
+			if err != nil {
+				foundConditions = append(foundConditions, operatorv1.OperatorCondition{
+					Type:    "WellKnownNotReadyProgressing",
```

Comment:
`WellKnownProgressing` and `WellKnownAvailable` please, otherwise this is just confusing

Reply:
@stlaz this matches the existing condition names... unfortunately we can't remove them otherwise during upgrade if they are set they will never change... we might think of a stale condition controller that will allow us to change these.